src/components/timeline/TimelineContextMenu.svelte

```diff
@@ -0,0 +1,120 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import type { ScaleTime } from 'd3-scale';
+  import { createEventDispatcher } from 'svelte';
+  import { view } from '../../stores/views';
+  import type { ActivityDirective, ActivityDirectivesMap } from '../../types/activity';
+  import type { Span, SpansMap, SpanUtilityMaps } from '../../types/simulation';
+  import type { MouseOver, VerticalGuide } from '../../types/timeline';
+  import { getAllSpansForActivityDirective, getSpanRootParent } from '../../utilities/activities';
+  import { getDoyTime, getIntervalInMs, getUnixEpochTimeFromInterval } from '../../utilities/time';
+  import { createVerticalGuide } from '../../utilities/timeline';
+  import ContextMenu from '../context-menu/ContextMenu.svelte';
+  import ContextMenuItem from '../context-menu/ContextMenuItem.svelte';
+  import ContextSubMenuItem from '../context-menu/ContextSubMenuItem.svelte';
+
+  export let activityDirectivesMap: ActivityDirectivesMap;
+  export let spansMap: SpansMap;
+  export let spanUtilityMaps: SpanUtilityMaps;
+  export let planStartTimeYmd: string;
+  export let contextMenu: MouseOver | undefined;
+  export let verticalGuides: VerticalGuide[];
+  export let xScaleView: ScaleTime<number, number> | null = null;
+
+  const dispatch = createEventDispatcher();
+  let contextMenuComponent: ContextMenu;
+  let activityDirective: ActivityDirective;
+  let activityDirectiveSpans: Span[] = [];
+  let span: Span;
+
+  // TODO imports here could be better, should we handle the vertical guide creation in Timeline?
+  $: timelines = $view.definition.plan.timelines;
+
+  $: if (contextMenu && contextMenuComponent) {
+    const { e, selectedActivityDirectiveId, selectedSpanId } = contextMenu;
+    console.log(e, selectedActivityDirectiveId, selectedSpanId);
+    contextMenuComponent.show(e);
+    activityDirective = null;
+    span = null;
+    activityDirectiveSpans = null;
+    if (typeof selectedActivityDirectiveId === 'number') {
+      activityDirective = activityDirectivesMap[selectedActivityDirectiveId];
+      activityDirectiveSpans = getAllSpansForActivityDirective(selectedActivityDirectiveId, spansMap, spanUtilityMaps);
+      activityDirectiveSpans.sort((a, b) => {
+        return getIntervalInMs(a.start_offset) < getIntervalInMs(b.start_offset) ? -1 : 1;
```

Comment:
Should we just have `getAllSpansForActivityDirective` sort as well? I don't _think_ we will ever want them out of order.

Reply:
Yeah could turn it into something that sorts activity directives and spans since they both have start offsets? And then apply that inside of `getAllSpansForActivityDirective`?

Reply:
I'd be ok with that

Reply:
Done